src/client/content_cao.cpp

```diff
@@ -824,39 +818,43 @@ void GenericCAO::updateLightNoCheck(u8 light_at_pos)
 
 	if (li != m_last_light)	{
 		m_last_light = li;
-		video::SColor color(255,li,li,li);
+		setNodeLight(li);
+	}
+}
 
-		if (m_enable_shaders) {
-			scene::ISceneNode *node = getSceneNode();
+void GenericCAO::setNodeLight(u8 light)
+{
+	video::SColor color(255, light, light, light);
 
-			if (node == nullptr) {
-				return;
-			}
+	if (m_enable_shaders) {
+		scene::ISceneNode *node = getSceneNode();
 
-			if (m_prop.visual == "upright_sprite") {
-				scene::IMesh *mesh = m_meshnode->getMesh();
-				for (u32 i = 0; i < mesh->getMeshBufferCount(); ++i) {
-					scene::IMeshBuffer* buf = mesh->getMeshBuffer(i);
-					video::SMaterial& material = buf->getMaterial();
-					material.EmissiveColor = color;
-				}
-			} else {
-				for (u32 i = 0; i < node->getMaterialCount(); ++i) {
-					video::SMaterial& material = node->getMaterial(i);
-					material.EmissiveColor = color;
-				}
+		if (node == nullptr)
+			return;
+
+		if (m_prop.visual == "upright_sprite") {
+			scene::IMesh *mesh = m_meshnode->getMesh();
+			for (u32 i = 0; i < mesh->getMeshBufferCount(); ++i) {
+				scene::IMeshBuffer *buf = mesh->getMeshBuffer(i);
+				video::SMaterial &material = buf->getMaterial();
+				material.EmissiveColor = color;
 			}
 		} else {
-			if (m_meshnode) {
-				setMeshColor(m_meshnode->getMesh(), color);
-			} else if (m_animated_meshnode) {
-				setAnimatedMeshColor(m_animated_meshnode, color);
-			} else if (m_wield_meshnode) {
-				m_wield_meshnode->setColor(color);
-			} else if (m_spritenode) {
-				m_spritenode->setColor(color);
+			for (u32 i = 0; i < node->getMaterialCount(); ++i) {
+				video::SMaterial &material = node->getMaterial(i);
+				material.EmissiveColor = color;
 			}
 		}
+	} else {
+		if (m_meshnode) {
+			setMeshColor(m_meshnode->getMesh(), color);
+		} else if (m_animated_meshnode) {
```

Comment:
maybe, without { }?

Reply:
If it's not enforced by the linter, I think it's better to leave this as it was before.

Reply:
Well, our style guide would probably say omit the braces.

Reply:
Feel free to suggest such code style in newly written code. For moved/re-indented code such trivial changes are not required IMO.

Reply:
This is ok as it is, uisng braces increases clarity.
Our code style guidelines do not insist on omitting the braces for single-line actions, either is allowed.

Reply:
Also, it is important to remember that the linter often makes very bad suggestions, only do what the linter suggests if it makes sense and is readable code, otherwise whitelist the file to silence the linter.